File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        di={}
        for i in range(len(nums)):
            remaining=target-nums[i]
            if remaining in di:
                return [di[remaining],i]
            di[nums[i]]=i
        

File: Data Structures & Algorithms/maximum-subarray/submission-0.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max_so_far=nums[0];
        int current=0;
        for(int val:nums)
        {
            current+=val;
            if(current>max_so_far)
            {
                max_so_far=current;
            }
            if(current<0)
            {
                current=0;
            }
        }
        return max_so_far;
        
    }
}
